mantidproject/mantid
# Muon Analysis: Wiring up fitting tab

**Description of work.**

This PR creates a backend for the fitting tab so that it can perform fits and put the results into the ADS. 

**To test:**
  * To access the Muon Analysis interface you need to add Muon Analysis 2 to the list of python interfaces in your user.properties file.
  * In mantidplot/workbench open the interface and load some data, MUSR22725 is in the external test data and should work for this purpose.
  * Go to the fitting tab, add the ExpDecayOsc fitting function from the Muon section and click fit. PLot the resulting workspace and check it looks sensible.
  * Change to sequential fit, click the select data to fit button, click select all and then select. 
  * Click fit again and confirm that a fit is done for each of the workspaces selected.
  * Select the simultaneous fit combo box, click fit and check the ADS for the results of the simultaneous fit.

**Known issues (which I'd ideally like to argue can be more efficiently done in a different PR's)**
  * Feedback on fit status
  * Need to add plotting after fit complete
  * TFA Asymmetry currently missing
  * Drag lines for fitting range selection



Fixes #25585 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
